rustls/src/server/verifier_builder.rs

```diff
@@ -0,0 +1,274 @@
+use std::sync::Arc;
+
+use crate::server::UnparsedCertRevocationList;
+use crate::verify::{AnonymousClientPolicy, ClientCertVerifier, WebPkiClientVerifier};
+use crate::{CertRevocationListError, RootCertStore};
+
+/// A builder for configuring a `webpki` client certificate verifier.
+///
+/// For more information, see the [`WebPkiClientVerifier`] documentation.
+#[derive(Debug, Clone)]
+pub struct ClientCertVerifierBuilder {
+    roots: RootCertStore,
+    crls: Vec<webpki::OwnedCertRevocationList>,
+    anon_policy: AnonymousClientPolicy,
+}
+
+impl ClientCertVerifierBuilder {
+    pub(crate) fn new(roots: RootCertStore) -> Result<Self, RootsRequiredError> {
+        match roots.is_empty() {
```

Comment:
So I think we should actually defer this error to `build()` time, because one of the downsides of the current approach is that there are potentially two places to deal with errors, both `builder()` and `with_crls()` -- and ideally the caller would want to handle those at the same time.

Reply:
I can see the advantage to only needing to handle error results once. Switched to this approach.